### PR TITLE
Simplified C++ interface for sox_io's get_info_file()

### PR DESCRIPTION
**Pull request by prabhat00155**

### CODE REVIEW

**Comment by prabhat00155** [comment]
Fixes https://github.com/pytorch/audio/issues/1219.

File: torchaudio/backend/sox_io_backend.py [hunk 1]
```diff
@@ -16,17 +16,10 @@ def _info(
...
-        sinfo.get_sample_rate(),
-        sinfo.get_num_frames(),
-        sinfo.get_num_channels(),
-        sinfo.get_bits_per_sample(),
-        sinfo.get_encoding(),
-    )
+    sinfo = (torchaudio._torchaudio.get_info_fileobj(filepath, format)
+             if hasattr(filepath, 'read') else
+             torch.ops.torchaudio.sox_io_get_info(os.fspath(filepath), format))
+    return AudioMetaData(*sinfo)
```
Comment (mthrok) [inline torchaudio/backend/sox_io_backend.py:22]: Can you try merge `_info` func and `info` in the same way as `load` function?

I think now TorchScript compiler can handle it without extracting the Python-only part to a helper function.

You can run the test locally with `pytest test/torchaudio_unittest/sox_io_backend/torchscript_test.py`

**Review by mthrok — APPROVED** [review]
Looks good, thanks. Could you try merging `_info` and `info` functions and see if it works? If it does not, let me know and I will merge this as-is.

**Comment by mthrok** [comment]
Thanks!

**Pull request merged**